Server/Main.cs

```diff
@@ -523,7 +541,7 @@ public static void Main(string[] args)
 
 				Console.WriteLine(" - Press return to exit, or R to try again.");
 
-				if (Console.ReadKey(true).Key != ConsoleKey.R)
+				if (Insensitive.Equals(Console.ReadLine(), "R"))

```

Comment:
This is user interaction which is desirable. We're not replacing Core.Service just extending it's functionality and making it so that a standard console tied to the IO streams is not needed.